torchaudio/compliance/kaldi.py

```diff
@@ -512,3 +513,213 @@ def fbank(
+
+    weights = torch.zeros_like(delta_t)
+    inside_window_indices = delta_t.abs().lt(window_width)
+    # raised-cosine (Hanning) window with width `window_width`
+    weights[inside_window_indices] = 0.5 * (1 + torch.cos(2 * math.pi * lowpass_cutoff /
+                                            lowpass_filter_width * delta_t[inside_window_indices]))
+
+    t_eq_zero_indices = delta_t.eq(0.0)
+    t_not_eq_zero_indices = ~t_eq_zero_indices
+    # sinc filter function
+    weights[t_not_eq_zero_indices] *= torch.sin(
+        2 * math.pi * lowpass_cutoff * delta_t[t_not_eq_zero_indices]) / (math.pi * delta_t[t_not_eq_zero_indices])
+    # limit of the function at t = 0
+    weights[t_eq_zero_indices] *= 2 * lowpass_cutoff
+
+    weights /= orig_freq  # size (output_samples_in_unit, max_weight_width)
+    return min_input_index, weights
+
+
+def _lcm(a, b):
+    return abs(a * b) // math.gcd(a, b)
+
+
+def _get_num_LR_output_samples(input_num_samp, samp_rate_in, samp_rate_out):
+    """ Based on LinearResample::GetNumOutputSamples. LinearResample (LR) means that
+    the output signal is at linearly spaced intervals (i.e the output signal has a
+    frequency of `new_freq`). It uses sinc/bandlimited interpolation to upsample/downsample
+    the signal.
+
+    Args:
+        input_num_samp (int): the number of samples in the input
+        samp_rate_in (float): the original frequency of the signal
+        samp_rate_out (float): the desired frequency
+
+    Returns:
+        int: the number of output samples
+    """
+    # For exact computation, we measure time in "ticks" of 1.0 / tick_freq,
+    # where tick_freq is the least common multiple of samp_rate_in and
+    # samp_rate_out.
+    tick_freq = _lcm(samp_rate_in, samp_rate_out)
+    ticks_per_input_period = tick_freq // samp_rate_in
+
+    # work out the number of ticks in the time interval
+    # [ 0, input_num_samp/samp_rate_in ).
+    interval_length_in_ticks = input_num_samp * ticks_per_input_period
+    if interval_length_in_ticks <= 0:
+        return 0
+    ticks_per_output_period = tick_freq // samp_rate_out
+    # Get the last output-sample in the closed interval, i.e. replacing [ ) with
+    # [ ].  Note: integer division rounds down.  See
+    # http://en.wikipedia.org/wiki/Interval_(mathematics) for an explanation of
+    # the notation.
+    last_output_samp = interval_length_in_ticks // ticks_per_output_period
+    # We need the last output-sample in the open interval, so if it takes us to
+    # the end of the interval exactly, subtract one.
+    if last_output_samp * ticks_per_output_period == interval_length_in_ticks:
+        last_output_samp -= 1
+    # First output-sample index is zero, so the number of output samples
+    # is the last output-sample plus one.
+    num_output_samp = last_output_samp + 1
+    return num_output_samp
+
+
+def resample_waveform(wave, orig_freq, new_freq, lowpass_filter_width=6):
```

Comment:
There was a comment about changing the name or parameters to emphasize the underlying algorithm use, LinearResample. What was the conclusion of it?

Reply:
We add comments/documentation to say what kind of resampling algorithm is used

Reply:
Is it expected that another algorithm could be used later?

Reply:
Not for compliance.kaldi but the transform layer has an argument for resampling method